### DISCUSSION TITLE
Multiple Sub-Sets, User-Sets (Prestige) on each Game Page

### DISCUSSION THREAD

**Discussion by ghost** [RetroAchievements/RAWeb#742] [Ideas]
It is planned to put Core, Bonus and Multiplayer all in one Game Page so the Users can earn all Achievements at same time and the Games don't need to be patched anymore.

Additional to this planned change, i suggest to split the Achievements on a Game Page into following:

1. **Core** (RA Approved Achievements)
**Minimum Requirement:** Dev Status
**Includes:** Easy-Hard Achievements
Default Set, every good player should be able to complete it.

2. **Sub-Sets** (RA Approved Achievements)
**Minimum Requirement:** Dev Status
**Includes:** Very Hard, Multiplayer, Extra Content (Game Version & Hack Specific)
This would replace the traditional Bonus/Multiplayer Set and allow us to being able to separate specific ingame content like Gold Saucer from Final Fantasy 7 to have its own Sub-Set. Sub-Sets would also allow to make Prototype or unused content Hack Achievements (only when the RAM addresses are identical to Core or it will be incompatible). Every Sub-Set gets his own Completion Badge in Users Profile.

3. **User-Sets** (Community Achievement Packs)
**Minimum Requirement:** Dev-Jr Status
**Includes:** Everything except duplicates that already exist in Core or a Sub-Set.
This would be a place where everyone can make Achievements he wants without breaking any rules. User-Sets with a RA Quality Seal (voted by Community) can be Prestige (increase Prestige counter by number of completed User-Sets for each Game) which will be displayed in Users Profile. There is no Completion Badge in Users Profile and Prestige counter has no limit. User-Sets need to be activated manually on the Game Page or Users Profile and only one User-Set can be active at same time to avoid earning multiple Prestige User-Sets at same time.

**Additional functions:**
- block Core-Set from being loaded with a Prototype and unused content Hacks MD5
- being able to change a Achievement from Core (Move) <-> Sub (Move) <-> User (Clone)
- vote function for User-Sets to get the RA Quality Seal to be turned into a Prestige Sets

**Comment by meleu** [comment]
Related topics:

- https://github.com/RetroAchievements/RAWeb/issues/143
- https://github.com/RetroAchievements/RAWeb/issues/105
- https://github.com/RetroAchievements/RASuite/issues/74

**Comment by Sanaki** [comment]
On the frontend level, the way I envision this would be to provide badges for core sets that act as a requisite for display of subsets, then allow augments to the badge for completing them. The main three augment types I'd envision as possible (in any combination) would be emblems (small icons that appear over the badge in specified locations), medals (small icons with mouseover identification that appear in unspecified order beneath the badge), and frames (probably just one possible per set, and likely reserved for bonus).

To give a specific example, let's take the generation 1 pokemon games, red and blue. You'd have the core set trigger the badge on your profile, which itself would have (for this example two) blank white spaces on the badge image akin to the empty positions for gym badges within the games. Completing red with any extra game-specific achievements would put a red emblem in the left one, and completing blue with any extras would put a blue emblem in the right one. Completing the bonus set would put a fancy frame around the badge itself, with either a gold rim or a secondary frame image associated with hardcore. Any further subcategories, such as multiplayer sets, minor hacks, and so forth would generate medals in a row (or possibly multiple rows if that many subcategories are even supported) beneath the badge tallying them up.

Assuming this idea were even implemented to its full degree, not all games would require all possible elements. Games that only have a core set would continue to function as they currently do. A game could have a bonus frame without emblems, or emblems without a bonus frame. I think visually it would be appealing to the player though, while soothing the issues some people have from both sides with set merges.

**Comment by Zenjir0** [comment]
I like this idea as well.

My take on it was a bit more simplistic, were we just have overlays for the badges, which can include all sorts of imaginary (heck could have special overlays for each game, system, series, etc), or just to keep it simple, use what we already have and change up the colors for the different levels of completion. Using Sanaki's example of Pokemon Red and Blue:

Gold (What we already use):
![020225_gold](https://user-images.githubusercontent.com/27794756/72862989-fdde7200-3c9c-11ea-88ed-7489eb22113a.png)

Silver (Complete specified achieves with hardmode enabled):
![020225_silver](https://user-images.githubusercontent.com/27794756/72863005-0a62ca80-3c9d-11ea-80f0-f51c762f9655.png)

Bronze (Complete specified achieves with hardmode disabled):
![020225_bronze](https://user-images.githubusercontent.com/27794756/72863037-2b2b2000-3c9d-11ea-8210-0aba43fea499.png)

if we wanted to add more layers we can add Platinum and Diamond. 

I will also throw in the idea of having animated badges (.gif) for complete mastery of all bonus achieves. 

The possibilities seem endless.

**Comment by televandalist** [comment]
Seeing "overlay" got me thinking about how transparent images got enabled a while back. That seems more fitting for the old "sticker" system that was proposed some time ago, though.

**Comment by Zenjir0** [comment]
Also I want to redact my initial idea and expand upon [JuliaSalsa](https://github.com/RetroAchievements/RAWeb/issues/143)'s design.

https://user-images.githubusercontent.com/35509989/40679355-2fced0c4-6383-11e8-8206-dadf92d29a4a.png

I say we add hollow or blank stars, and this will allow players to finish sub-sets in different orders. Much like in Super Mario 64, you could go for Star 4 or 6, even though the game/system wanted you to get Star 1. Another example is something like [FTL](https://cdn.cloudflare.steamstatic.com/steam/apps/212680/ss_9dd5cc63efcc5964d45713b7304e71d41be6a3d5.1920x1080.jpg?t=1589331228) where as you can see in the screenshot on the left side it has 3 locks. You don't have to complete them in order, but it is possible to unlock achievement C before unlocking A. Or just forgo the blank star idea and just tack stars on as certain requirements are met. But we should inform the user/player of what those are. Essentially universal achievements for every game such as; beating the main game, beating the main game with hardmode enabled, completing the full core-set, completing the challenge-set, etc 

I do believe we should have a simple "I beat the Game" badge/star. Let me give an example using Super Mario Bros 3 (SMB3) for the NES. 
In the game you could speedrun it by using the warp whistles to get to world 8 bypassing all the other worlds, and Complete the game. This should award 1 star minimum. However, unlocking the completion achievement for each world would unlock star 2, and completing all the extra achievements such as spawning the white mushrooms, winning the memory game, scoring X amount of points within X amount of time, etc would unlock the 3rd star. Something else to think of is multiple star unlocks such as. Beating the game does unlock one star, but beating the game with hardmode enabled will unlock 2, one for simply beating the game, the other for having hardmode enabled.
